backend/dataall/cdkproxy/stacks/environment.py

```diff
@@ -734,7 +675,7 @@ def create_group_environment_role(self, group):
 
         group_role = iam.Role(
             self,
-            f'{group.environmentIAMRoleName}',
+            id,
```

Comment:
If you wonder why I added this --> it is for backwards compatibility. 

Original environment - before this PR
EnvironmentA with Admin = Scientists and InvitedTeam = Engineers. In the environment stack we will create 2 roles `...-environmentA-...` CloudFormation Id ="EnvAdminRole" for the Scientists team and `...-engineers-..` with CloudFormation Id  ...-engineers-..` for Engineers.

Reply:
Upgraded environment - after this PR
We do not want IAM roles to be renamed, because that would delete roles and create new ones, which will break previous Lake Formation and S3 permissions. But also the CloudFormation resource Id needs to be the same as the original one, otherwise CloudFormation will try to create a new resource and that will cause issues with the old role that gets cleaned up after the CloudFormation deployment